src/ui.rs

```diff
@@ -253,6 +256,17 @@ impl<T> Snarl<T> {
                 let draw_order = self.draw_order.clone();
                 let mut drag_released = false;
 
+                let mut center = vec2(0.0, 0.0);
+                let mut need_centering = false;
+                // Centering nodes
+                if style.centering{
+                    ui.input(|i|{
+                        if i.pointer.button_double_clicked(PointerButton::Primary){
```

Comment:
This will cause centering even if clicked to a node or a pin, or even outside the graph.
If you intend to make centering on double-click to the background, check it where panning is checked.